src/term/check/unbound_vars.rs

```diff
@@ -4,7 +4,7 @@ use std::collections::HashMap;
 
 impl DefinitionBook {
   pub fn check_unbound_vars(&self) -> anyhow::Result<()> {
-    for def in &self.defs {
+    for (_, def) in &self.defs {
```

Comment:
You can do `for def in self.defs.values()` to iterate here. Same goes for the other identical `for`s.